x-pack/plugins/infra/public/pages/metrics/hosts/components/table/filter_action.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiPopover, EuiButtonEmpty } from '@elastic/eui';
+
+const selectedHostsLabel = (selectedHostsCount: number) => {
+  return i18n.translate('xpack.infra.hostsViewPage.table.selectedHostsButton', {
+    values: { selectedHostsCount },
+    defaultMessage:
+      'Selected {selectedHostsCount} {selectedHostsCount, plural, =1 {host} other {hosts}}',
+  });
+};
+
+interface HostsTableFilterActionProps {
+  selectedItemsCount: number;
+  filterSelectedHosts: () => void;
+}
+
+export const HostsTableFilterAction = ({
+  selectedItemsCount,
+  filterSelectedHosts,
+}: HostsTableFilterActionProps) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
```

Comment:
nit: @mohamedhamed-ahmed just showed me this `useBoolean` hook, it's pretty cool for cases like this. We could do this instead:

```ts
const [isPopoverOpen, { off: closePopover, toggle: togglePopover }] = useBoolean(false);
```

and use `off` and `togglePopover` instead of `setIsPopoverOpen`